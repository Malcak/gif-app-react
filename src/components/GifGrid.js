import React from 'react';
import PropTypes from 'prop-types';
import { useFechGifs } from '../hooks/useFechGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {
  const { data: images, loading } = useFechGifs(category);

  return (
    <>
      <h3>{category}</h3>
      {loading && <p>Loading...</p>}
      <div className="card-grid">
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
