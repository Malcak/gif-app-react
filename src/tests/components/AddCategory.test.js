import React from 'react';
import { shallow } from 'enzyme';

import { AddCategory } from '../../components/AddCategory';

describe('tests in <AddCategory /> component', () => {
  const setCategories = jest.fn();
  let wrapper = shallow(<AddCategory setCategories={setCategories} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  test('should display the <AddCategory /> component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('should change the text input', () => {
    const value = 'Hello World';
    wrapper.find('input').simulate('change', { target: { value: value } });
    expect(wrapper.find('input').prop('value')).toBe(value);
  });
  test('the posting of the information on submit should fail', () => {
    wrapper.find('form').simulate('submit', { preventDefault() {} });
    expect(setCategories).not.toHaveBeenCalled();
  });
  test('should call the setCategories function and clear the text input', () => {
    const value = 'Hello World';
    const input = wrapper.find('input');
    input.simulate('change', { target: { value: value } });
    wrapper.find('form').simulate('submit', { preventDefault() {} });
    expect(setCategories).toHaveBeenCalled();
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
    expect(input.prop('value')).toBe('');
  });
});
