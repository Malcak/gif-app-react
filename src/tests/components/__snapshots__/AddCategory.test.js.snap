// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in <AddCategory /> component should display the <AddCategory /> component correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
